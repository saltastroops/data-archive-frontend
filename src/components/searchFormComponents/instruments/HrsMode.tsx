import * as React from "react";
import MultiSelectField, {
  AllOption,
} from "../../basicComponents/MultiSelectField";

interface IHrsModeProps {
  hrsModes?: string[];
  onSelect: (value: any) => void;
}

const HRS_MODES = [
  "Low Resolution",
  "Medium Resolution",
  "High Resolution",
  "High Stability",
  "Int Cal Fibre",
];
/**
 * A form for selecting HrsMode-related search parameters.
 */
const HrsMode = (props: IHrsModeProps) => {
  const { onSelect, hrsModes } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      hrsModes: Array.from(values).map((t: any) => {
        return t.text;
      }),
    });
  };

  return (
    <div>
      <p>HRS mode</p>
      <MultiSelectField
        name={"hrsMode"}
        onChange={select}
        value={hrsModes || ["All"]}
      >
        <AllOption />
        {HRS_MODES.sort().map((i) => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default HrsMode;
