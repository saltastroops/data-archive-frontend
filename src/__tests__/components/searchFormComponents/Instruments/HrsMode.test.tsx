import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import HrsMode from "../../../../components/searchFormComponents/instruments/HrsMode";

const onSelect = jest.fn() as any;
describe("Hrs Mode", () => {
  it("should be defined", () => {
    expect(mount(<HrsMode hrsModes={[]} onSelect={onSelect} />)).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(mount(<HrsMode hrsModes={[]} onSelect={onSelect} />))
    ).toMatchSnapshot();
  });
});
