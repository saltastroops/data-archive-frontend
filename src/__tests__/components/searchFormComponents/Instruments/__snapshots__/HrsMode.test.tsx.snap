// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hrs Mode should render correctly 1`] = `
.c0 {
  width: 95%;
}

<HrsMode
  hrsModes={Array []}
  onSelect={[MockFunction]}
>
  <div>
    <p>
      HRS mode
    </p>
    <MultiSelectField
      name="hrsMode"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <div
          className="c0 control select is-multiple "
        >
          <select
            multiple={true}
            name="hrsMode"
            onChange={[Function]}
            size={6}
            style={
              Object {
                "width": "216px",
              }
            }
            value={Array []}
          >
            <AllOption>
              <option
                value="All"
              >
                All
              </option>
            </AllOption>
            <option
              key="High Resolution"
              value="High Resolution"
            >
              High Resolution
            </option>
            <option
              key="High Stability"
              value="High Stability"
            >
              High Stability
            </option>
            <option
              key="Int Cal Fibre"
              value="Int Cal Fibre"
            >
              Int Cal Fibre
            </option>
            <option
              key="Low Resolution"
              value="Low Resolution"
            >
              Low Resolution
            </option>
            <option
              key="Medium Resolution"
              value="Medium Resolution"
            >
              Medium Resolution
            </option>
          </select>
        </div>
      </styled.div>
    </MultiSelectField>
  </div>
</HrsMode>
`;
